pegasus/sites.v3/code.org/views/course_table.haml

```diff
@@ -286,3 +264,102 @@
+
+
+%h2 Tools
+
+%p Code.org has also produced a variety of tools that can be used standalone.  They all originally appear in the K-12 curriculum.
+
+.tools.courses
+  - [0,1].each do |row|
+    - tools_for_row = tools[row*6, row*6+6]
+    - tools_for_row.each_with_index do |tool, index|
+      - tool_index = row * 6  + index
+      - width = tool[:layout] == "wide" ? 240 : 100
+      .tool{id: "tool-#{index}", style: "width: #{width + 40}px; float: left; padding: 10px; text-align: center; cursor: pointer", data: { index: tool_index, row: row } }
+        %img{src: tool[:img] || "/shared/images/fill-#{width}x100/courses/logo_tall_course1.jpg", style: "border-radius: 10px;"}
+        %br/
+        .name{style: "font-size: 12px; font-family: 'Gotham 5r', sans-serif; line-height: 14px; padding-top: 4px"}
+          = tool[:name]
+
+    .detailrow{id: "detailrow-#{row}", style: "display: none; position: relative; clear: both; height: 140px; "}
+      - tools_for_row.each_with_index do |tool, index|
+        - tool_index = row * 6 + index
+        - width = tool[:layout] == "wide" ? 240 : 100
+        .course.detailcontainer{id: "toolsextra-#{tool_index}", style: "display: none; color: grey; cursor: auto; position: absolute; top: 0px; left: 0px"}
+          .arrow_box{class: "tools-arrow-offset-#{index}"}
+            .arrow_box_content{style: "overflow: hidden"}
+              - if tool[:img]
+                .left{style: "float: left"}
+                  %a{href: tool[:link]}
+                    %img.courseimage{src: tool[:img], style: "width: #{width}px"}
+              .right{style: "float: left"}
+                .title
+                  =tool[:name]
+                - if tool[:long]
+                  .description
+                    =tool[:long]
+                - else
+                  .description
+                    =tool[:description]
+                - if tool[:link]
+                  %br/
+                  %a{href: tool[:link], style: "padding-left: 0px; color: grey"}
+                    Learn more
+            .tool_arrow_box_close{style: "position: absolute; right: 10px; top: 0px; cursor: pointer", onclick: "$('#toolsextra-#{tool_index}').slideUp()", data: { index: index } }
+              %i.fa.fa-times{"aria-hidden" => "true"}
+
+
+:javascript
+  var detailRowShowing = -1;
+  var toolShowing = -1;
+
+  $(document).ready()
+  {
+    $('.tool').click(function() {
+      var row = ($(this).data('row'));
+      var index = ($(this).data('index'));
+
+      if (toolShowing == -1) {
+        $('#detailrow-' + row).slideDown();
+        $('#toolsextra-' + index).fadeIn();
+        detailRowShowing = row;
+        toolShowing = index;
+      } else if (toolShowing == index) {
+        $('#detailrow-' + detailRowShowing).slideUp();
+        $('#toolsextra-' + index).fadeOut();
+        detailRowShowing = -1;
+        toolShowing = -1;
+      } else if (detailRowShowing == row) {
```

Comment:
Is `detailRowShowing` only for courses section while `toolShowing` is only for tools section? Could use a comment or two?